libvast/builtins/operators/unflatten.cpp

```diff
@@ -0,0 +1,77 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "vast/argument_parser.hpp"
+#include "vast/pipeline.hpp"
+#include "vast/table_slice.hpp"
+
+#include <vast/concept/parseable/string/char_class.hpp>
+#include <vast/concept/parseable/vast/pipeline.hpp>
+#include <vast/error.hpp>
+#include <vast/logger.hpp>
+#include <vast/plugin.hpp>
+
+#include <arrow/type.h>
+
+namespace vast::plugins::unflatten {
+
+namespace {
+
+constexpr auto default_unflatten_separator = ".";
+
+class unflatten_operator final : public crtp_operator<unflatten_operator> {
+public:
+  unflatten_operator() = default;
+
+  unflatten_operator(std::string separator) : separator_{std::move(separator)} {
+  }
+
+  auto operator()(generator<table_slice> input,
+                  [[maybe_unused]] operator_control_plane& ctrl) const
+    -> generator<table_slice> {
+    for (auto&& slice : input) {
+      auto result = vast::unflatten(slice, separator_);
+      co_yield std::move(result);
+    }
+  }
+
+  auto to_string() const -> std::string override {
+    return fmt::format("unflatten '{}'", separator_);
+  }
+
+  auto name() const -> std::string override {
+    return "unflatten";
+  }
+
+  friend auto inspect(auto& f, unflatten_operator& x) -> bool {
+    return f.apply(x.separator_);
+  }
+
+private:
+  std::string separator_ = default_unflatten_separator;
+};
+
+class plugin final : public virtual operator_plugin<unflatten_operator> {
+public:
+  auto parse_operator(parser_interface& p) const -> operator_ptr override {
+    auto parser
+      = argument_parser{"head", "https://vast.io/next/"
```

Comment:
```suggestion
      = argument_parser{"unflatten", "https://vast.io/next/"
```

Reply:
Fixed in https://github.com/tenzir/tenzir/pull/3304/commits/50a4ca3a39b0c885e41cd85e91723a405ae5de82